### PR TITLE
fix: Don't fail if doc column is missing

### PR DESCRIPTION
**Pull request by erichare**
This pull request improves the robustness and usability of the `extract_docling_documents` function in `docling_utils.py`, especially when working with DataFrames that may not have the expected column names. It also adds comprehensive unit tests to ensure correct behavior and helpful error messaging in various scenarios.

**Enhancements to DataFrame extraction logic:**

* Added a fallback mechanism in `extract_docling_documents` to search for columns containing `DoclingDocument` objects when the exact column name is not found, and logs a warning if a fallback is used.
* Improved error messages to provide users with actionable suggestions when the expected column is missing, including listing available columns and possible solutions.

**Testing improvements:**

* Added a full suite of unit tests for `extract_docling_documents`, covering extraction from `Data`, lists of `Data`, DataFrames with correct and incorrect columns, fallback behavior, and error handling for empty inputs and missing columns.

## Summary by CodeRabbit

* **New Features**
  * Added intelligent fallback mechanism for document extraction: system now searches available columns when the specified column is not found, with informative guidance.

* **Bug Fixes**
  * Improved error messages to provide detailed information about available columns and recommended actions for extraction issues.

* **Tests**
  * Comprehensive unit test coverage added covering normal operations, edge cases, and error scenarios for document extraction.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/lfx/src/lfx/base/data/docling_utils.py [hunk 1]
```diff
@@ -32,14 +32,48 @@ def extract_docling_documents(data_inputs: Data | list[Data] | DataFrame, doc_ke
...
+                        found_column = col
+                        break
+                except (IndexError, AttributeError):
+                    continue
+
+            if found_column:
+                logger.warning(
+                    f"Column '{doc_key}' not found, but found DoclingDocument objects in column '{found_column}'. "
+                    f"Using '{found_column}' instead. Consider updating the 'Doc Key' parameter."
+                )
```
Comment (ogabrielluiz) [inline src/lfx/src/lfx/base/data/docling_utils.py]: We need to surface this to the UI

Comment (erichare) [reply src/lfx/src/lfx/base/data/docling_utils.py]: @ogabrielluiz how does it look now?
